dev/docker-compose.yaml

```diff
@@ -6,7 +6,6 @@ x-airflow-common:
   build:
     context: ..
     dockerfile: dev/Dockerfile
-  env_file: .env
```

Comment:
@CorsettiS ATM, the `.env` file exposes sensitive information to the container, such as passwords and API keys. That's why it is not committed in the main repo, but we expect users to be able to use it.

Reply:
but for local development, if not needed, is there any advantage in keeping it in the docker compose yaml file?

Reply:
What would be the alternative for the developers who want to use the `.env`? Would they need to change the `docker-compose.yaml` file?

Reply:
I would advocate for that, since from my perspective the ideal scenario for local development is the one that you face zero problems to set it up.   The idea of the `docker-compose.yaml` file is to simply create a test dev for contributors, and If someone wants to use the `.env`, it makes sense for me that the person should set it up starting from the minimum provided setup.
This is just my opinion as a dev tho.

Reply:
I agree with zero problems during setup.

I wonder if creating an empty `.env` would be better than completely removing it. There may also be other alternatives - do you have any other suggestions?

As a dev, considering you have sensitive connection information (e.g., DB password), how do you currently make it available in `docker-compose-yml`?

I've seen several developers using the `.env` approach. If we removed this support, several developers would have problems with the local setup, which this PR aims to avoid.

Reply:
Creating an empty `.env` sounds like a reasonable approach for me, although it would be needed to remove it from the .gitignore file.

If I have any sensitive info I have that I want to test out in airflow (such as DB passwords) I usually pass it directly to the airflow UI, which I thought was the default way that devs were doing it.

Reply:
@CorsettiS We find the `.env` file particularly useful when testing against multiple versions of Python and Airflow - it makes it much quicker to set sensitive information without worrying about the UI.

A proposal:
- have an empty `.env` in the repo (we'd remove it from `.gitignore`)
- have a pre-commit hook to block users from committing a non-empty `.env`

Reply:
@tatiana  done. I have tested locally and it works fine.